coalib/coala.py

```diff
@@ -59,26 +59,56 @@ def main(debug=False):
 
         if args.show_bears:
             from coalib.settings.ConfigurationGathering import (
-                get_filtered_bears)
-
-            local_bears, global_bears = get_filtered_bears(
-                args.filter_by_language, log_printer)
-
-            show_bears(local_bears,
-                       global_bears,
-                       args.show_description or args.show_details,
-                       args.show_details,
-                       console_printer)
-
+                get_all_bears)
+            all_bears = get_all_bears(log_printer)
+            if args.filter_by_language:
+                # filter_by_language option should be deprecated soon.
+                # Since it is already covered by filter_by option.
+                log_printer.warn(
+                    "'--filter-by-language ...' option is deprecated. "
+                    "Use '--filter-by language ...' option instead")
+                if args.filter_by is None:
+                    args.filter_by = []
+                filter = ['language']
+                filter.extend(args.filter_by_language)
+                args.filter_by.append(filter)
+            if args.filter_by:
+                # Each iteration of the following loop applies
+                # filters one by one provided as arguments
+                for filter_by in args.filter_by:
+                    from coalib.parsing.Filter import Filter
+                    filter = filter_by[0]
+                    filter_args = filter_by[1:]
+                    try:
+                        all_bears = Filter.get_filtered_bears(
+                            filter, filter_args, all_bears)
+                    except Exception as ex:
+                        console_printer.print(ex)
+                        return 2
+
+                local_bears, global_bears = all_bears
+                show_bears(local_bears,
+                           global_bears,
+                           args.show_description or args.show_details,
+                           args.show_details,
+                           console_printer)
+            else:
+                # To print all the bears if no filters provided
+                local_bears, global_bears = all_bears
+                show_bears(local_bears,
+                           global_bears,
+                           args.show_description or args.show_details,
+                           args.show_details,
+                           console_printer)
```

Comment:
the code in the else branch looks the same like in the normal "if" branch. The loop won't enter if there are no filters, so imo you can safely remove the `if`